alarm_trap_sender.hpp

```diff
@@ -61,40 +61,43 @@ class AlarmListEntry
   std::string _issuer;
 };
 
-// Iterator for enumerating entries in the active alarm list. Subclassed
+// Iterator for enumerating entries in the ObservedAlarms mapping. Subclassed
 // from map iterator to hide pair template. Only operations defined are
 // supported.
 
-class ActiveAlarmIterator : public std::map<unsigned int, AlarmListEntry>::iterator
+class ObservedAlarmsIterator : public std::map<unsigned int, AlarmListEntry>::iterator
 {
 public:
-  ActiveAlarmIterator(std::map<unsigned int, AlarmListEntry>::iterator iter) : std::map<unsigned int, AlarmListEntry>::iterator(iter) {}
+  ObservedAlarmsIterator(std::map<unsigned int, AlarmListEntry>::iterator iter) : std::map<unsigned int, AlarmListEntry>::iterator(iter) {}
 
   AlarmListEntry& operator*()  {return  (std::map<unsigned int, AlarmListEntry>::iterator::operator*().second);}
   AlarmListEntry* operator->() {return &(std::map<unsigned int, AlarmListEntry>::iterator::operator*().second);}
 };
 
-// Container for all currently active alarms. Currently indexed by alarm
-// model index (may need to be extended to include a resource id going
-// forward). 
+// Container for all alarms we have seen being raised (either at CLEARED or
+// non-CLEARED severity). Maps the index of an alarms to the latest severity
+// with which it was raised. Currently indexed by alarm model index (may need 
+// to be extended to include a resource id going forward). 
 
-class ActiveAlarmList
+class ObservedAlarms
 {
 public:
-  ActiveAlarmList() {}
+  ObservedAlarms() {}
 
-  // Adds a non CLEARED severity alarm to the list if it does not already
-  // exist. For a CLEARED severity update, removes an associated alarm of
-  // the same alarm model index if one exists. Returns true if a change
-  // was made to the list.
+  // Adds a non-CLEARED severity alarm to the mapping if it does not already
+  // exist and returns true. If an entry does exist but at a different severity
+  // then we update the mapping and return true. For a CLEARED severity update,
+  // if the alarm doesn't exist in the mapping or if it does exist but at a
+  // non-CLEARED severity, we update the mapping and return true.
```

Comment:
It sounds like the processing for Cleared and non-cleared seventies are basically the same.